### PR TITLE
drop stackdriver metrics and tracing exporters

### PR DESCRIPTION
**Pull request by dprotaso**
- drop stackdriver as a tracing backend
- drop stackdriver as a metrics backend
- update deps - dropping stackdriver

Fixes: https://github.com/knative/pkg/issues/2173

### CODE REVIEW

File: metrics/config_test.go [hunk 1]
```diff
@@ -30,27 +27,14 @@ import (
...
-
 const (
-	servingDomain          = "knative.dev/serving"
-	internalServingDomain  = "knative.dev/internal/serving"
-	eventingDomain         = "knative.dev/eventing"
-	internalEventingDomain = "knative.dev/internal/eventing"
-	customSubDomain        = "test.domain"
-	testProj               = "test-project"
-	anotherProj            = "another-project"
+	metricsDomain = "knative.dev/project"
```
Comment (skonto) [inline metrics/config_test.go:37]: What was the value of having different domains?

**Comment by skonto** [comment]
@dprotaso I still see some references to the `stackdriver` name like here: https://github.com/knative/pkg/blob/952ee7db67049390a6a1067657370b711400d433/logging/config.go#L58
Shouldn't we remove it from the whole codebase?

**Comment by dprotaso** [comment]
re:
```go
loggingCfg := stackdriverConfig() 
```

This just produces a logging format that stackdriver can parse easily - I'm ok with leaving this since it helps debugging with our CI/prow runs. 

Do you have suggestions on other alternatives here?

File: metrics/config_test.go (hunk 1 above)
Comment (dprotaso) [reply metrics/config_test.go:37]: This was a stack driver thing but we're actually prefixing this to the open census metric names so I think this should keep this around until we resolve: https://github.com/knative/pkg/issues/2174


I think there's some value in repurposing this as a label in order to help disambiguate between components across different Knative projects (ie. serving vs eventing).

ie. in the future we drop the metric prefix and thus require labels to help segment metrics.

So for a golang memory metric - we'd have a component label ie. `controller`  and reuse the domain label `serving`

I'm ok to _drop_ this domain if an alternative exists - ie. if the components namespace is automatically picked up as part of the metric

**Comment by skonto** [comment]
> Do you have suggestions on other alternatives here?
I can do a followup PR to cleanup anything that remains and discuss them in detail one by one. For that particular logging config func I would just rename it, if it does not do any harm.

**Comment by dprotaso** [comment]
/hold 

I'll fix the downstream serving issues

**Comment by dprotaso** [comment]
Downstream issue should be resolved here:
https://github.com/knative/serving/pull/11659

**Comment by dprotaso** [comment]
Moving Eventing Source StatReporter out of knative.dev/pkg - https://github.com/knative/eventing/pull/5587

**Comment by dprotaso** [comment]
I'm going to temporarily move the eventing sources metric keys to `pkg/source` until we can move the stats reporter into the eventing repo.

**Comment by dprotaso** [comment]
last PR to fix serving downstream: https://github.com/knative/serving/pull/11679

File: metrics/config_observability.go [hunk 2]
```diff
@@ -65,8 +65,8 @@ type ObservabilityConfig struct {
 	// EnableProbeRequestLog enables queue-proxy to write health check probe request logs.
 	EnableProbeRequestLog bool
 
-	// RequestMetricsBackend specifies the request metrics destination, e.g. Prometheus,
-	// Stackdriver. "None" disables all backends.
+	// RequestMetricsBackend specifies the request metrics destination, e.g. Prometheus.
+	// "None" disables all backends.
```
Comment (evankanderson) [inline metrics/config_observability.go:69]: Should this have OpenCensus?

File: metrics/config_test.go (hunk 1 above)
Comment (evankanderson) [reply metrics/config_test.go:37]: I think the namespace is likely to be picked up as part of the metric; I'm not sure if we have to / want to worry about cases like queue-proxies and sources running in a user's namespace.

I'm inclined to think that namespace + some sort of component/Resource name should be sufficient here, but I might be mistaken.

Comment (evankanderson) [reply metrics/config_test.go:37]: (With the notion that the Resource would indicate whether it was a serving or eventing construct if it was running in a user namespace, and that multi-tenant components would be running in a namespace where the name would give a clue.)

The obvious counter-example here is `mink`, where all the controllers are bundled into a single Pod, but I'm not sure that's a compelling enough case.

File: metrics/config_test.go [hunk 3]
```diff
@@ -835,7 +507,6 @@ func TestUpdateExporter(t *testing.T) {
 func TestUpdateExporterFromConfigMapWithOpts(t *testing.T) {
 	setCurMetricsConfig(nil)
 	oldConfig := getCurMetricsConfig()
-	successTestsInit()
```
Comment (evankanderson) [inline metrics/config_test.go:838]: Do we still need `successTestsInit`? I've seen it removed several places here.

**Review by evankanderson — COMMENTED** [review]
/lgtm
/approve

I'm amused that we're deleting about 10k lines of `aws-sdk-go` with this change. (Also, wow the generated proto files stackdriver pulled in were big...)

File: metrics/config_test.go (hunk 3 above)
Comment (dprotaso) [reply metrics/config_test.go:838]: It's removed - in the diff

File: metrics/config_observability.go (hunk 2 above)
Comment (dprotaso) [reply metrics/config_observability.go:69]: I'll add it back

**Pull request merged**